### ISSUE TITLE
Question: AWS ElastiCache Security Group ID create new resources?

### ISSUE THREAD

**Issue by stack72** [hashicorp/terraform#2303]
I have the following:

```
resource "aws_security_group" "redis" {
  name = "redis-sg"
  description = "ElastiCache Redis Securtity Group"
    vpc_id = "${aws_vpc.default.id}"

    ingress {
      from_port = "6739"
      to_port = "6739"
      protocol = "tcp"
      security_groups = ["${aws_security_group.node.id}"]
    }

    egress {
      from_port = "0"
      to_port = "0"
      protocol = "-1"
      cidr_blocks = ["0.0.0.0/0"]
    }

    tags {
      Name = "ElastiCache Redis Node"
    }
}

resource "aws_elasticache_subnet_group" "sensu_redis_csng" {
  name = "csng"
  description = "Redis Cache Subnet Group"
  subnet_ids = ["${aws_subnet.primary-private.id}","${aws_subnet.secondary-private.id}","${aws_subnet.tertiary-private.id}"]
}

resource "aws_elasticache_cluster" "sensu_redis_cache" {
  cluster_id = "redis"
  engine = "redis"
  port = 6379
  engine_version = "2.8.19"
  node_type = "cache.m3.medium"
  num_cache_nodes = 1
  subnet_group_name = "${aws_elasticache_subnet_group.sensu_redis_csng.name}"
  security_group_ids = ["${aws_security_group.redis.id}"]
}

```
On changing the security_group_ids of the aws_elasticache_cluster, a new resource will be forced (according to the image below)
Should security group id's for an ElastiCache cluster *really* force a new resource?

![screen shot 2015-06-10 at 10 31 34](https://cloud.githubusercontent.com/assets/227823/8079398/0bc9531c-0f5c-11e5-8764-82ae8ea070d5.png)

Ideally, changing Security Groups should be a non-destructive action.

**Comment by catsby** [comment]
No, it _should_ not force a new resource. Unfortunately the current code doesn't really do much in the [update method][1] :frowning: 

There are many things we should allow updating with [ModifyCacheCluster][2]

I'll put this on my list but I don't think I'll get to it anytime this week.
If anyone else wants to take a stab at it, please make mention here, and feel free to ask any questions here. Otherwise I'll try to get to this next week, hopefully 😕

[1]: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_elasticache_cluster.go#L233
[2]: http://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_ModifyCacheCluster.html

**Comment by stack72** [comment]
@catsby thanks for the response - this isn't urgent at all. I was just curious as to whether this was actually the way it works or not

thanks for the response :)

**Comment by catsby** [comment]
Closing this now; #2469 enables in-place updates so things like changing a security group doesn't require a re-create (inside a VPC, at least).

Thanks for reporting!

**Comment by stack72** [comment]
@catsby this is great - thanks :)

**Comment by catsby** [comment]
Just merged #2469

**Comment by garlou** [comment]
>  ingress {
>       from_port = "6739"
>       to_port = "6739"
>       protocol = "tcp"
>       security_groups = ["${aws_security_group.node.id}"]
>     }

>   port = 6379

I know it has been a long time but I'm struggling here with elasticache tf configurations and I wanted to ask, are the ports switched on purpose?
